gateway/reverse_proxy.go

```diff
@@ -787,6 +785,40 @@ func (rt *TykRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
 	return rt.transport.RoundTrip(r)
 }
 
+const (
+	inMemNetworkName = "in-mem-network"
+	inMemNetworkType = "memu"
+)
+
+func handleInMemoryLoop(handler http.Handler, r *http.Request) (resp *http.Response, err error) {
+	// use separate provider for each request
+	provider := memconn.Provider{}
+
+	// start in mem listener
+	lis, err := provider.Listen(inMemNetworkType, inMemNetworkName)
+	// on lis.Close http.Serve will return with error and stop goroutine
+	defer func() { _ = lis.Close() }()
+
+	// start http server with in mem listener
+	// Note: do not try to use http.Server it is working only with mux
+	mux := http.NewServeMux()
+	mux.Handle("/", handler)
+	go func() { _ = http.Serve(lis, mux) }()
```

Comment:
Where you stop it?
Feels like potential memory leak.

Reply:
Close it after `client.Do`?

Reply:
It will be stopped after closing listener

```
// on lis.Close http.Serve will return with error and stop goroutine
defer func() { _ = lis.Close() }()
```